pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -28,58 +28,95 @@ import (
 	errorutils "k8s.io/apimachinery/pkg/util/errors"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	clientset "k8s.io/client-go/kubernetes"
-	appslisters "k8s.io/client-go/listers/apps/v1"
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/client-go/util/retry"
 )
 
-// StatefulPodControlInterface defines the interface that StatefulSetController uses to create, update, and delete Pods,
+// StatefulPodControlObjectManager abstracts the manipulation of Pods and PVCs. The real controller implements this
+// with a clientset and listers; for test indexers are used instead.
+type StatefulPodControlObjectManager interface {
```

Comment:
That interface change is necessary to refactor the unittests, so that StatefulSetControl can actually be unittested. Before, the fake pod control it used duplicated logic from the real implementation and so was not actually testing the pod control (see the deleted lines 1734 in statful_set_control_test).

I think it would have been more clear to make that refactoring in a separate commit, but that was hard to split out from the work in progress.